testing/test_model.py

```diff
@@ -52,5 +52,35 @@ def test_film_generator():
     assert (type(inf[0]) == torch.Tensor)
     assert (type(inf[1]) == torch.nn.parameter.Parameter)
 
+def test_modified_encoder():
+    # verifying changes in the Encoder class
```

Comment:
is it really what this test verifies? currently it only checks if `inf` is a `tuple`

Reply:
yes, it was strange for me at first, but if we attend to the output of the encoder, we will find out that we need a list that contains features that represent skip connection of Unet network. 
furthermore, at the final line of code in the encoder class, a film layer joins this list, and these two features compacted as a tuple.

Reply:
> if we attend to the output of the encoder, we will find out that we need a list that contains features that represent skip connection of Unet network.
furthermore, at the final line of code in the encoder class, a film layer joins this list, and these two features compacted as a tuple.

right, but at the end you are only assessing if your output is a tuple, your are not assessing its _integrity_

Reply:
note: the same comment applies to your 2 other tests

Reply:
I think that these three tests evaluate the  _integrity_  assessment too. I added the unit test for encoder and decoder sections separately to ensure that some changes made by me will not change the behavior of the codebase. In the third unit test which is *unet_att_2d* the _integrity_ assessment is done because it consists of the encoder, decoder, and attention module.

using the separate unit test for encoder and decoder class helped me to find out in the bottom layer of the encoder section, the number of filters did not double in the [bottom layer](https://github.com/ivadomed/ivadomed/pull/552#discussion_r536899063)

Reply:
@jcohenadad my current problem is that I don't understand the error messages which surprisingly were crystal clear last night 😃, 
here are current error messages:
*FAILED testing/test_HeMIS.py::test_HeMIS - RuntimeError: Given groups=1, weig...
*FAILED testing/test_training_time.py::test_unet_time[config1-target_lst0-train_lst0]
*FAILED testing/test_training_time.py::test_unet_time[config5-target_lst0-train_lst0]
So, now I don't have any idea about why these errors show up while I just modify the test_model.py not test_HeMIS nor the test_training_time

Reply:
I suggest you run pytest locally under the debugger

Reply:
I tested it locally and upload the related Jupiter notebook file [here](https://github.com/AshkanTaghipour/Attention_2D_Unet/blob/main/2D_attention_Unet.ipynb) that shows the validity of the 2d attention module on Unet. 
However, I will do it on PyCharm too.